qiskit/quantum_info/analysis/z2_symmetries.py

```diff
@@ -0,0 +1,471 @@
+                of the of single-qubit list, {len(sq_list)}."
+            )
+
+        if tapering_values is not None:
+            if len(sq_list) != len(tapering_values):
+                raise QiskitError(
+                    f"The length of the single-qubit list, {len(sq_list)}, must match the length of the \
+                    tapering values, {len(tapering_values)} ."
+                )
+
+        self._symmetries = symmetries
+        self._sq_paulis = sq_paulis
+        self._sq_list = sq_list
+        self.tapering_values = tapering_values
+        self.tol = tol
+
+    @property
+    def symmetries(self):
+        """Return symmetries."""
+        return self._symmetries
+
+    @property
+    def sq_paulis(self):
+        """Return sq paulis."""
+        return self._sq_paulis
+
+    @property
+    def cliffords(self) -> list[SparsePauliOp]:
+        """
+        Get clifford operators, build based on symmetries and single-qubit X.
+
+        Returns:
+            A list of unitaries used to diagonalize the Hamiltonian.
+        """
+        cliffords = [
+            (SparsePauliOp(pauli_symm) + SparsePauliOp(sq_pauli)) / np.sqrt(2)
+            for pauli_symm, sq_pauli in zip(self._symmetries, self._sq_paulis)
+        ]
+        return cliffords
+
+    @property
+    def sq_list(self):
+        """Return sq list."""
+        return self._sq_list
+
+    @property
+    def settings(self) -> dict:
+        """Return operator settings."""
+        return {
+            "symmetries": self._symmetries,
+            "sq_paulis": self._sq_paulis,
+            "sq_list": self._sq_list,
+            "tapering_values": self.tapering_values,
+        }
+
+    def __str__(self):
+        ret = ["Z2 symmetries:"]
+        ret.append("Symmetries:")
+        for symmetry in self._symmetries:
+            ret.append(symmetry.to_label())
+        ret.append("Single-Qubit Pauli X:")
+        for x in self._sq_paulis:
+            ret.append(x.to_label())
+        ret.append("Cliffords:")
+        for c in self.cliffords:
+            ret.append(str(c))
+        ret.append("Qubit index:")
+        ret.append(str(self._sq_list))
+        ret.append("Tapering values:")
+        if self.tapering_values is None:
+            possible_values = [
+                str(list(coeff)) for coeff in itertools.product([1, -1], repeat=len(self._sq_list))
+            ]
+            possible_values = ", ".join(x for x in possible_values)
+            ret.append("  - Possible values: " + possible_values)
+        else:
+            ret.append(str(self.tapering_values))
+
+        ret = "\n".join(ret)
+        return ret
+
+    def is_empty(self) -> bool:
+        """
+        Check the z2_symmetries is empty or not.
+
+        Returns:
+            Empty or not.
+        """
+        return len(self._symmetries) == 0 or len(self._sq_paulis) == 0 or len(self._sq_list) == 0
+
+    # pylint: disable=invalid-name
```

Comment:
It is nice time to rename the method now. What do you think? @mrossinek 
For example find_z2_symmetries

Reply:
Sure we can do that :+1: 

I figured the pylint exception is also due to some of the internally used variable names but we can adapt those too :+1:

Reply:
A proposal for renaming methods with pylint errors:
- find_Z2_symmetries -> find_z2_symmetries() or find_symmetries() (in the end Z2Symmetries.find_symmetries() seems descriptive enough to avoid the redundancy)
- _kernel_F2() -> _kernel_f2() or  _kernel_binary_field()
- _row_echelon_F2() -> _kernel_f2() or _row_echelon_binary_field()

Reply:
For now I'm going with the lowercase which is the easiest choice.